app/src/main/java/com/duckduckgo/app/browser/favicon/FaviconManager.kt

```diff
@@ -21,163 +21,212 @@ import android.net.Uri
 import android.widget.ImageView
 import androidx.core.net.toUri
 import com.duckduckgo.app.bookmarks.db.BookmarksDao
+import com.duckduckgo.app.bookmarks.model.FavoritesRepository
 import com.duckduckgo.app.browser.favicon.FileBasedFaviconPersister.Companion.FAVICON_PERSISTED_DIR
 import com.duckduckgo.app.browser.favicon.FileBasedFaviconPersister.Companion.FAVICON_TEMP_DIR
 import com.duckduckgo.app.browser.favicon.FileBasedFaviconPersister.Companion.NO_SUBFOLDER
 import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteRepository
 import com.duckduckgo.app.global.DispatcherProvider
 import com.duckduckgo.app.global.domain
 import com.duckduckgo.app.global.faviconLocation
+import com.duckduckgo.app.global.touchFaviconLocation
+import com.duckduckgo.app.global.view.loadFavicon
 import com.duckduckgo.app.location.data.LocationPermissionsRepository
 import kotlinx.coroutines.withContext
 import java.io.File
 
 interface FaviconManager {
-    suspend fun saveToTemp(subFolder: String, icon: Bitmap, url: String): File?
-    suspend fun prefetchToTemp(subFolder: String, url: String): File?
-    suspend fun persistFavicon(subFolder: String, url: String)
-    suspend fun loadFromTemp(subFolder: String, url: String): Bitmap?
-    suspend fun loadToViewFromTemp(subFolder: String, url: String, view: ImageView)
-    suspend fun loadToViewFromPersisted(url: String, view: ImageView)
+    suspend fun storeFavicon(tabId: String, faviconSource: FaviconSource): File?
```

Comment:
as discussed offline, I think these changes could have been done in a separate PR